expands mobile page access

thurt opened the pull request:
add mobile .tpl files;
modify mobile/header.tpl to accomodate;
add mobile php files;
fix template paths to mobile/ and path to initialize.php;
add mobile-table CSS class;
change main mobile/ pages to use mobile-table class;
fix path to subm_bg.gif;
remove Top Menu (the one with black backgroud);
remove Charts, Reports, Invoices, Export from submenu;
remove fixed width for many input elements;
remove nowrap attribute in header.tpl submenu;
add inline style for background-repeat in header.tpl submenu;
prevent textarea's from being resizable horizontally, allow them to be resizable vertically;
remove fixed width from textarea cols attribute;
add mobile-textarea CSS class;
fix bug in project_edit.php -- onload event referring to wrong property name;
change expenses date input to be like that of time date input;

thurt commented:
This PR is a WIP

thurt commented:
I'm totally flexible about this PR. I wanted to post the visuals of what I have so far in order to get your input @anuko. So let me know your ideas.

I started off by copying the regular page .php and .tpl files into their repsective mobile/ folders and then modifying them from there.

I decided Reports, Charts, Invoices, Export should be omitted from mobile because I consider these tasks to be administrative or analysis, which is separate from daily input work. Another idea would be to allow the user to modify which pages are available via their profile settings.

A question: I was trying to add a mobile-only CSS class to the textboxes like
```php
$form->addInput(array('type'=>'textarea','name'=>'description','class'=>'mobile-textarea','value'=>$cl_description));
```
but the class doesn't seem to get added on the output end? I haven't figured out why yet.

anuko commented:
There is one issue that I noticed while testing. The mobile/users.php may not fit on screen when login is long, for example, when users use their emails as logins. One possible solution is to not display the Edit and Delete columns, and instead show an optional link  for user name or login, which would be a user edit link, from where they also should be able to delete a user.

thurt commented:
> The mobile/users.php may not fit on screen when login is long, for example, when users use their emails as logins

Good point. I found the PR has the same issue with long project names also. There may be some way to fix this no matter what the length of text is with `table-layout: fixed` and `word-wrap: break-word` CSS rules. I haven't experimented with this much yet, but here's the idea: http://stackoverflow.com/questions/5889508/using-word-wrap-break-word-within-a-table

> One possible solution is to not display the Edit and Delete columns, and instead show an optional link for user name or login

Yea that's a good idea to save some more space. I will try to change that.

thurt commented:
>  There may be some way to fix this no matter what the length of text is with table-layout: fixed and word-wrap: break-word CSS rules

Adding a CSS rule `word-break: break-all` to just the first two columns (for Name and Login) is the most straightforward way to fix the issue without affecting the other columns. Here is a sample with an egregiously long name/login. 

![word-break-320w](https://cloud.githubusercontent.com/assets/8955144/17903437/76021006-6939-11e6-85da-e40858ebc068.png)

thurt commented:
Note: `word-break: break-all` has a downside here. The effect on long multi-word descriptions, you can see it fills up the maximum horizontal space in the cell before breaking the word instead of breaking between words. That maybe isn't as nice looking. But I'm not sure how to accommodate for both long single words (which need to have `break-all` rule) and long multi-word descriptions (which look best with `break-word` rule).

![word-break-desc](https://cloud.githubusercontent.com/assets/8955144/17904084/eaf15f6e-693b-11e6-80bf-c8d79e7e874b.png)

thurt commented:
`overflow-wrap: break-word` (aka `word-wrap`) does accommodate both long single words and long multi-word descriptions, but the parent `table` must have `table-layout: fixed`. I think this is the best way to go, so I will start a commit for this. But it does introduce some new issues.

The current 10% widths on the the td's does not look nice with `table-layout: fixed`. 10% / 320px (the lower bound of resolution for mobile) is 32px. 32px - 1px cellspacing is 31 px which is not enough room for 10pt verdana in columns Role and Delete.

![tt-users](https://cloud.githubusercontent.com/assets/8955144/17940082/732a7338-69fb-11e6-9ee7-1dc72f02cbbc.png)

I will fix this by freeing up some horizontal space by first removing the Delete column and put the Delete functionality in the Edit page.

anuko commented:
When testing this pacth, I am seeing a presentation problem on the Users page in the Role column. Am I missing something, or the patch does not contain everything? See the picture below.

![role_display_problem](https://cloud.githubusercontent.com/assets/2672997/17942916/efd8a86e-6a28-11e6-8cd1-d0142b4324ca.png)

thurt commented:
Probably need to also include `word-wrap: break-word;`. `word-wrap` and `overflow-wrap` are the same rule, but this link indicates that some older browsers only know `word-wrap`

https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap

thurt commented on mobile/expense_edit.php:
```diff
@@ -26,7 +26,7 @@
 // +----------------------------------------------------------------------+
 
-require_once('initialize.php');
+require_once('../initialize.php');
```
I just forgot to correct the paths for this one page
